File: src/GloboTicket.Web/src/types/venue.ts
/**
 * Venue interface matching the backend VenueDto
 */
export interface Venue {
  /** Unique identifier for the venue */
  id: number;
  
  /** Unique GUID identifier for the venue */
  venueGuid: string;
  
  /** Name of the venue */
  name: string;
  
  /** Physical address of the venue */
  address?: string;
  
  /** Latitude coordinate of the venue location */
  latitude?: number;
  
  /** Longitude coordinate of the venue location */
  longitude?: number;
  
  /** Maximum seating capacity of the venue */
  seatingCapacity: number;
  
  /** Description of the venue */
  description: string;
  
  /** UTC timestamp when the venue was created */
  createdAt: string;
  
  /** UTC timestamp when the venue was last updated */
  updatedAt?: string;
}

/**
 * DTO for creating a new venue
 */
export interface CreateVenueDto {
  venueGuid: string;
  name: string;
  address?: string;
  latitude?: number;
  longitude?: number;
  seatingCapacity: number;
  description: string;
}

/**
 * DTO for updating an existing venue
 */
export interface UpdateVenueDto {
  name: string;
  address?: string;
  latitude?: number;
  longitude?: number;
  seatingCapacity: number;
  description: string;
}

File: src/GloboTicket.Domain/Entities/Venue.cs
using NetTopologySuite.Geometries;

namespace GloboTicket.Domain.Entities;

/// <summary>
/// Represents a venue where events can be held.
/// Includes location data and seating capacity information.
/// </summary>
public class Venue : MultiTenantEntity
{
    /// <summary>
    /// Gets or sets the unique identifier for this venue.
    /// </summary>
    public Guid VenueGuid { get; set; }

    /// <summary>
    /// Gets or sets the name of the venue.
    /// </summary>
    public string Name { get; set; } = string.Empty;

    /// <summary>
    /// Gets or sets the physical address of the venue.
    /// </summary>
    public string? Address { get; set; } = null;

    /// <summary>
    /// Gets or sets the geographic location of the venue as a point (longitude, latitude).
    /// Uses the WGS84 coordinate system (SRID 4326).
    /// </summary>
    public Point? Location { get; set; } = null;

    /// <summary>
    /// Gets or sets the maximum seating capacity of the venue.
    /// </summary>
    public int SeatingCapacity { get; set; } = 0;

    /// <summary>
    /// Gets or sets a description of the venue.
    /// </summary>
    public string Description { get; set; } = string.Empty;
}